src/renderer/components/+extensions/extensions.tsx

```diff
@@ -32,27 +54,238 @@ export class Extensions extends React.Component {
     return extensionManager.localFolderPath;
   }
 
+  getExtensionDestFolder(name: string) {
+    return path.join(this.extensionsPath, sanitizeExtensionName(name));
+  }
+
+  installFromSelectFileDialog = async () => {
+    const { dialog, BrowserWindow, app } = remote;
+    const { canceled, filePaths } = await dialog.showOpenDialog(BrowserWindow.getFocusedWindow(), {
+      defaultPath: app.getPath("downloads"),
+      properties: ["openFile", "multiSelections"],
+      message: _i18n._(t`Select extensions to install (formats: ${this.supportedFormats.join(", ")}), `),
+      buttonLabel: _i18n._(t`Use configuration`),
+      filters: [
+        { name: "tarball", extensions: this.supportedFormats }
+      ]
+    });
+    if (!canceled && filePaths.length) {
+      this.requestInstall(
+        filePaths.map(filePath => ({
+          fileName: path.basename(filePath),
+          filePath: filePath,
+        }))
+      );
+    }
+  };
+
+  installExtensions = () => {
+    if (this.downloadUrl) {
+      this.installFromNpmOrUrl(this.downloadUrl);
+      this.downloadUrl = "";
+    } else {
+      this.installFromSelectFileDialog();
+    }
+  };
+
+  installFromNpmOrUrl = async (url = this.downloadUrl) => {
+    if (!InputValidators.isUrl.validate(url)) {
+      const npmPackageName = url;
+      url = extensionManager.getNpmPackageTarballUrl(npmPackageName);
+      if (!url) {
+        Notifications.error(`Error: npm package "${npmPackageName}" not found!`);
+        return;
+      }
+    }
+    try {
+      const { promise: filePromise } = downloadFile({ url });
+      this.requestInstall([{
+        fileName: path.basename(url),
+        data: await filePromise,
+      }]);
+    } catch (err) {
```

Comment:
[nit] unnecessary abbreviation, e.g. "error" can be used instead

Reply:
What? :D I agree if it would be just `e`.
Maybe `evt` also unnecessary abbreviation for event-handlers all around and we should use `event` instead?

Reply:
I'd use `event` also. This is suggested in e.g. book "Code Complete", which has some empirical studies for it. Not saying this is super-important, thus marked as "nit". (For me "nit" means that it doesn't need to be fixed necessarily, but would be nice).